lib/performance-monitor.ts

```
@@ -0,0 +1,167 @@
+    return new Promise(async (resolve, reject) => {
+      const timerId = this.startTimer(name, metadata);
+      
+      try {
+        const result = await fn();
+        this.endTimer(timerId, metadata);
+        resolve(result);
+      } catch (error) {
+        this.endTimer(timerId, { ...metadata, error: true });
+        reject(error);
+      }
+    });
```

Biome (2.1.2)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)